src/io/EventReader.cc

```diff
@@ -50,6 +50,22 @@ EventReader::result_type EventReader::operator()()
 
         for (auto gen_particle : gen_event.particles())
         {
+            // Check if this particle type is present in the problem data and
+            // skip the particle if not
+            bool found = false;
+            for (auto md : params_->md())
+            {
+                if (gen_particle->pid() == md.pdg_code.get())
+                {
+                    found = true;
+                    break;
+                }
+            }
+            if (!found)
+            {
+                continue;
+            }
```

Comment:
Let's leave this out for now, since it would cause silent failures. If invalid particle types become an issue in our event input files, we could perhaps build a list of the invalid event IDs for user feedback. I suppose this linear search is OK since the number of particles is going to be small, but it would still be better written as an algorithm:
```suggestion
            PDGNumber pdg{gen_particle->pid()};
            if (std::find_if(
                    params_->md().begin(),
                    params_->md().end(),
                    [pdg](const ParticleMd& md) { return md.pdg_code == pdg; })
                == params_->md().end())
            {
                // Particle type is not defined for the current physics
                missing_pdg.insert(pdg);
                continue;
            }
```

And on thinking about it, I'd actually change `ParticleParams` so that `find` returns an unassigned `ParticleDefId` if not found, and it's the caller's responsibility to call `CHECK(def_id);` -- that'll make debugging easier as well as simplify this for you.